clash-lib/src/Clash/Driver/Types.hs

```diff
@@ -362,6 +390,35 @@ data ClashOpts = ClashOpts
   }
   deriving (Show)
 
+instance NFData ClashOpts where
+  rnf o =
+    opt_inlineLimit o `deepseq`
+    opt_specLimit o `deepseq`
+    opt_inlineFunctionLimit o `deepseq`
+    opt_inlineConstantLimit o `deepseq`
+    opt_evaluatorFuelLimit o `deepseq`
+    opt_cachehdl o `deepseq`
+    opt_clear o `deepseq`
+    opt_primWarn o `deepseq`
+    opt_color o `seq`
```

Comment:
We should probably add our own type with NFData..

Reply:
Yeah... there are some other GHC types we use directly which I think are better replaced with analogues in `clash-lib`, like `InlineSpec` and `SrcLoc`. Even if they're the same, using them directly from `ghc` is a weird coupling between the front-end and the back-end of the compiler IMO

Reply:
But it's a bit tangential, I'll save it for another PR. I think I'll spend the first few days next week doing some more _get ready for 1.6_ PRs